cmd/tfsec/main.go

```diff
@@ -169,7 +171,32 @@ var rootCmd = &cobra.Command{
 			os.Exit(1)
 		}
 
-		if allInfo(results) || softFail {
+		// Soft fail always takes precedence. If set, only execution errors
+		// produce failure exit codes.
+		if softFail {
+			os.Exit(0)
+		}
+
+		if detailedExitCode {
```

Comment:
Can we pull this logic into a function and return the exit code derived? Otherwise I like it